src/libs/Middleware/Reauthentication.js

```diff
@@ -58,13 +58,8 @@ function Reauthentication(response, request, isFromSequentialQueue) {
                 // There are some API requests that should not be retried when there is an auth failure like
                 // creating and deleting logins. In those cases, they should handle the original response instead
                 // of the new response created by handleExpiredAuthToken.
-                const shouldRetry = lodashGet(request, 'data.shouldRetry');
                 const apiRequestType = lodashGet(request, 'data.apiRequestType');
-                if (!shouldRetry && !apiRequestType) {
-                    if (isFromSequentialQueue) {
-                        return data;
-                    }
-
+                if (!isFromSequentialQueue && !apiRequestType) {
```

Comment:
Since we don't have shouldRetry anymore, I think it should just be this. 
```javascript
if (!apiRequestType) {
    if (request.resolve) {
        request.resolve(data);
    }
    return data;
}
```
If the request is from the SequentialQueue we want to return data here.

Reply:
we don't have shouldRetry, but we still retry requests from the sequential queue. So if a request from the sequential queue returns `ONST.JSON_CODE.NOT_AUTHENTICATED` we want to reauthenticate then retry the request.

Reply:
If you remove `!isFromSequentialQueue` tests that test reauthentication in the sequential queue will fail

Reply:
> if a request from the sequential queue returns `CONST.JSON_CODE.NOT_AUTHENTICATED` we want to reauthenticate then retry the request.

Ah ok. I see now. Since we remove shouldRetry from SequentialQueue requests we would fall into this block and not reauthenticate before retrying. Could you add some explanation like that as a comment please? Thanks for explaining it to me. This logic is confusing.